src/Adapter/MSTest.TestAdapter/ObjectModel/UnitTestElement.cs

```diff
@@ -230,6 +230,8 @@ internal TestCase ToTestCase()
 
     private void SetTestCaseId(TestCase testCase, string testFullName)
     {
+        testCase.SetPropertyValue(Constants.TestIdGenerationStrategyProperty, (int)TestMethod.TestIdGenerationStrategy);
```

Comment:
Saved and restored as int. I could use string which would be more human readable but int feels safer for renames.

Note that we are simply doing round-trip between discovery and run so it's probably as safe to be using string.

Reply:
can be saved somewhere and reused in the future? You know what I mean...could this break other usages? Like breaking change in the enum?

Reply:
After some investigation, I will keep this as `int` for now as it matches what's done for the other enums. Also, it seems that there are some issues with using `string`, I will try to investigate them.